### PR TITLE
2015 day 9 part 1

### PR DESCRIPTION
**Pull request by jdosornio**
Created lib to calculate the shortest path to travel all the existing locations only once

From:
https://adventofcode.com/2015/day/9

### CODE REVIEW

**Comment by jdosornio** [comment]
[2015-day-9-part-1](https://trello.com/c/xBBOxvTB/9-2015-day-9-part-1)

File: tests/day_4.rs [hunk 1]
```diff
@@ -4,22 +4,22 @@ use advent_of_code::year2015::day_4::get_minimum;
 
 #[test]
 #[ignore]
-fn test_day_4_one() {
+fn test_day_4_1() {
```
Comment (kanekv) [inline tests/day_4.rs]: merge one line tests into 1 section

File: src/year2015/day_9.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+use std::collections::HashSet;
+use regex::Regex;
+
+
+pub fn get_shortest_distance(lines: Vec<String>) -> u32 {
+    let mut nodes = HashSet::new();
+    let mut distances = HashMap::new();
+    let re = Regex::new(r"^(?P<start>\w+)\s*to\s*(?P<end>\w+)\s*=\s*(?P<dist>\d+)$")
+        .unwrap();
+
```
Comment (kanekv) [inline src/year2015/day_9.rs]: remove extra empty line

**Comment by jdosornio** [comment]
Done with the requested changes

**Pull request merged**